src/System.Net.Security/src/System/Net/SecureChannel.cs

```diff
@@ -1059,16 +1060,12 @@ internal SecurityStatusPal Encrypt(byte[] buffer, int offset, int size, ref byte
                 resultSize = 0;
 
                 int bufferSizeNeeded = checked(size + _headerSize + _trailerSize);
-                if (output != null && bufferSizeNeeded <= output.Length)
-                {
-                    writeBuffer = output;
-                }
-                else
+
```

Comment:
@CIPop I was originally going to move this calculation into the Windows PAL part, but wasn't sure how you'd feel about the exception logging changes.

I see that SslStreamPal(.Windows).EncryptMessage uses LoggingHash.HashString(securityContext) already; I could copy the catch { log+throw } block into that method and get rid of the extra consts; if that's good with you.